php/ext/google/protobuf/encode_decode.c

```diff
@@ -371,6 +369,21 @@ static void* str_handler(void *closure,
 }
 
 static bool str_end_handler(void *closure, const void *hd) {
+  stringfields_parseframe_t* frame = closure;
+  const upb_fielddef **field = hd;
+  MessageHeader* msg = (MessageHeader*)frame->closure;
+
+  CACHED_VALUE* cached = find_zval_property(msg, *field);
+
+  new_php_string(cached, frame->sink.ptr, frame->sink.len);
+
+  stringsink_uninit(&frame->sink);
+  free(frame);
```

Comment:
Is this safe?

Reply:
1) new_php_string copy the string
2) The frame is created in str_handler, which is always paired with str_end_handler